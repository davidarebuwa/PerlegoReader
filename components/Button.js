import React from "react";
import { Text, View, StyleSheet, Pressable } from "react-native";
import common from "../constants/styles";

export default function Button(props) {
  const { onPress, title = "Menu" } = props;
  return (
    <Pressable style={common.button} onPress={onPress}>
      <Text style={common.text}>{title}</Text>
    </Pressable>
  );
}
 


