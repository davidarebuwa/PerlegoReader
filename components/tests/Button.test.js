import React from "react";
import Button from "../Button";
import { render, fireEvent } from "@testing-library/react-native";

describe("Button", () => {
    it("should render correctly", () => {
        const { toJSON } = render(<Button />);
        expect(toJSON()).toMatchSnapshot();
    });
    
    it("should call onPress when pressed", () => {
        const onPress = jest.fn();
        const { getByTestId } = render(<Button onPress={onPress} />);
        fireEvent.press(getByTestId("button"));
        expect(onPress).toHaveBeenCalled();
    });
    });

    