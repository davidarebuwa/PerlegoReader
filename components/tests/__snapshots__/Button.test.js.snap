// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 4,
      "elevation": 3,
      "justifyContent": "center",
      "paddingHorizontal": 32,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "letterSpacing": 0.25,
        "lineHeight": 21,
      }
    }
  >
    Menu
  </Text>
</View>
`;
